### PR TITLE
Update release checklist template

### PR DESCRIPTION
**Pull request by swissspidy**
Based on yesterday's experience

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -3,14 +3,19 @@ name: "\U0001F680 Regular Release Checklist"
 about: "\U0001F512 Maintainers only: create a checklist for a regular release process"
 title: 'Release checklist for v2.x.x'
 labels: 'i: scope:distribution'
-assignees: 'schlessera'
+assignees: '@wp-cli/maintainers'
```
Comment (danielbachhuber) [inline .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:6]: I don't think this works, actually:

CleanShot 2023-11-01 at 15 38 48@2x

We could probably leave it empty.

File: .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md [hunk 2]
```diff
@@ -3,14 +3,19 @@ name: "\U0001F680 Regular Release Checklist"
...
 # Regular Release Checklist - v2.x.x
 
 ### Preparation
 
-- [ ] Draft release post on the [make.wordpress.org CLI blog](https://make.wordpress.org/cli/wp-admin/post-new.php)
+- [ ] Mention on Slack that a release is being prepared
+
+    People should wait with updating until the announcement. Before that, things are still in motion.
+
+- [ ] Verify results of [automated test suite](https://github.com/wp-cli/automated-tests)
```
Comment (danielbachhuber) [inline .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:17]:
```suggestion
- [ ] Verify all tests pass in the [automated test suite](https://github.com/wp-cli/automated-tests)
```

File: .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md [hunk 3]
```diff
@@ -116,50 +152,54 @@ assignees: 'schlessera'
     md5 hash verified: 179fc8dacbfe3ebc2d00ba57a333c982
     New version works. Proceeding to replace.
     Success: Updated WP-CLI to 1.4.0.
-    $ wp @daniel option get home
```
Comment (danielbachhuber) [inline .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:119]: This was meant as a smoke test...

File: .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md [hunk 4]
```diff
@@ -116,50 +152,54 @@ assignees: 'schlessera'
...
+- [ ] Optional: Announce using the `/announce` slash command in the [`#cli`](https://wordpress.slack.com/messages/C02RP4T41) Slack room.
+
+    This pings a lot of people, so it's not always desired. Plus, the blog post will pop up on Slack anyway.
 
 ### Bumping WP-CLI version again
 
 - [ ] Bump [VERSION](https://github.com/wp-cli/wp-cli/blob/master/VERSION) in [`wp-cli/wp-cli`](https://github.com/wp-cli/wp-cli) again.
 
-    For instance, if the release version was `0.24.0`, the version should be bumped to `0.25.0-alpha`. Doing so ensure `wp cli update --nightly` works as expected.
+    For instance, if the release version was `0.24.0`, the version should be bumped to `0.25.0-alpha`.
```
Comment (danielbachhuber) [inline .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:195]:
```suggestion
    For instance, if the release version was `2.8.0`, the version should be bumped to `2.9.0-alpha`. 
```

File: .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md (hunk 1 above)
Comment (swissspidy) [reply .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:6]: Hmm only works for reviewers then I guess

File: .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md (hunk 3 above)
Comment (swissspidy) [reply .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:119]: A smoke test that only works on your machine :-)

What‘s a good alternative?

Comment (schlessera) [reply .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:119]: How about:
```
$ wp cli version
WP-CLI 2.8.1
$wp eval 'echo \WP_CLI\Utils\http_request( "GET", "https://api.wordpress.org/core/version-check/1.6/" )->body;' --skip-wordpress
<PHP serialized string with version numbers>
```
That would at the very least test whether the update worked and whether regular execution and the integration with Requests works.
Note: The above example that Daniel provided mostly just tested the SSH subsystem. The entire option retrieval logic is executed on whatever version is installed on his server.

**Review by schlessera — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md (hunk 3 above)
Comment (danielbachhuber) [reply .github/ISSUE_TEMPLATE/4-REGULAR_RELEASE_CHECKLIST.md:119]: Added in c532c80a

**Pull request merged**